src/api/__tests__/api.unit.test.ts

```diff
@@ -1,81 +1,351 @@
 import "core-js";
 import mockAxios from "jest-mock-axios";
-import { API_ROOT, getApiUrl, apiGet } from "../api";
-import ERRORS from "../../shared/errors";
-import { makeTestError } from "../../shared/test-utils";
+import {
+  retrieveFactory,
+  getFactory,
+  createFactory,
+  patchFactory,
+  deleteFactory,
+  endpointFactory,
+  unwrap,
+} from "../api";
+import {
+  MOCK_RESPONSE,
+  MOCK_ERROR,
+  MOCK_BODY,
+  MOCK_CSRF,
+  MOCK_PARAMS,
+} from "../../shared/test-utils";
 
-describe("getApiUrl", () => {
-  it("returns root endpoint on empty string", () => {
-    expect(getApiUrl("")).toBe(API_ROOT);
+describe("listFactory", () => {
```

Comment:
(Applies to all)

Should have beforeEach hooks to clean up axios state. This way, you don't have to call mockResponseFor; you can just call mockResponse and the correct call will be chosen.

Alternatively, wrap everything in a describe (e.g. `describe("API factories")`) and place the beforeEach hook in that.

Reply:
Using `mockResponseFor` is more about ensuring we're hitting the right URLs. Unless there's a more idiomatic way to do that I'm missing?

Reply:
You still want to clean up axios state after each test case, since otherwise changes persist across test cases (and with Jest, you can't expect your cases to run in order).

You raise a good point with the mockResponseFor; I say we keep that.

Reply:
I thought I was already doing the `mockAxios.reset` thing, but apparently I've just been very lucky. https://github.com/UWmathNEWS/slugline-web/pull/33/commits/edc74c0ed0842941fc32cafeea12669b1a32a48f